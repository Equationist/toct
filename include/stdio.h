/* stdio.h - Standard Input/Output Library for CompilerKit C Compiler */

#ifndef _STDIO_H
#define _STDIO_H

/* Standard I/O types */
typedef struct {
    int _fd;
    int _flags;
    char* _buffer;
} FILE;

/* Standard streams */
extern FILE* stdin;
extern FILE* stdout; 
extern FILE* stderr;

/* Standard I/O functions */
int printf(const char* format, ...);
int fprintf(FILE* stream, const char* format, ...);
int sprintf(char* str, const char* format, ...);
int snprintf(char* str, int size, const char* format, ...);

int scanf(const char* format, ...);
int fscanf(FILE* stream, const char* format, ...);
int sscanf(const char* str, const char* format, ...);

int putchar(int c);
int puts(const char* s);
int fputs(const char* s, FILE* stream);
int fputc(int c, FILE* stream);

int getchar(void);
char* gets(char* s);
char* fgets(char* s, int size, FILE* stream);
int fgetc(FILE* stream);
int ungetc(int c, FILE* stream);

FILE* fopen(const char* filename, const char* mode);
FILE* freopen(const char* filename, const char* mode, FILE* stream);
int fclose(FILE* stream);
int fflush(FILE* stream);

long ftell(FILE* stream);
int fseek(FILE* stream, long offset, int whence);
void rewind(FILE* stream);
int feof(FILE* stream);
int ferror(FILE* stream);
void clearerr(FILE* stream);

/* File position constants */
#define SEEK_SET 0
#define SEEK_CUR 1
#define SEEK_END 2

/* End of file constant */
#define EOF (-1)

/* Buffer size */
#define BUFSIZ 8192

/* Null pointer */
#ifndef NULL
#define NULL ((void*)0)
#endif

#endif /* _STDIO_H */