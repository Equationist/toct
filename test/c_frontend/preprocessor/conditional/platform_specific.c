/* Test platform-specific conditional compilation */
#include <stdio.h>

int main() {
    printf("Platform detection:\n");

#ifdef __linux__
    printf("Running on Linux\n");
#elif defined(__APPLE__) && defined(__MACH__)
    printf("Running on macOS\n");
#elif defined(_WIN32) || defined(_WIN64)
    printf("Running on Windows\n");
#else
    printf("Unknown platform\n");
#endif

#ifdef __GNUC__
    printf("Compiled with GCC version %d.%d\n", __GNUC__, __GNUC_MINOR__);
#elif defined(__clang__)
    printf("Compiled with Clang\n");
#elif defined(_MSC_VER)
    printf("Compiled with MSVC\n");
#else
    printf("Unknown compiler\n");
#endif

#if defined(__x86_64__) || defined(_M_X64)
    printf("64-bit architecture\n");
#elif defined(__i386) || defined(_M_IX86)
    printf("32-bit architecture\n");
#else
    printf("Unknown architecture\n");
#endif

    return 0;
}