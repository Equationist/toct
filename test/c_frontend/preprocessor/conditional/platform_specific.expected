Platform detection:
Running on macOS
Compiled with (GCC version [0-9]+\.[0-9]+|Clang)
64-bit architecture